src/application-constants.ts

```diff
@@ -141,3 +141,7 @@ export const GITHUB_ENVIRONMENT_VARIABLES = {
 }
 export const GITHUB_TOKEN_VALIDATION_SARIF_UPLOAD_ERROR = 'Missing required GitHub token for uploading SARIF report to GitHub Advanced Security'
 export const SARIF_REPORT_ERROR_FOR_PR_SCANS = 'SARIF report create/upload is ignored in case of PR scans, it is only supported for non PR scans'
+export const POLARIS_PR_COMMENT_ERROR_FOR_NON_PR_SCANS = 'Polaris PR comment is enabled but it will be ignored since it is not a PR scan'
```

Comment:
1. simplify the error messages like: Polaris PR comment ignored for non pull request scan.
2. as discussed mandatory validations should be executed first eg. if bd token is not available we should error out and not go to pr comment flow at all. take care for similar use cases.

Reply:
updated